### ISSUE TITLE
Deserializing xml on linux coreclr x64 throws System.NotSupportedException

### ISSUE THREAD

**Issue by IRooc** [dotnet/corefx#2174]
When executing the following code on ubuntu 14.04 using coreclr x64 version 1.0.0-beta6-12120 a System.NotSupportedException is thrown. This works fine on windows with same coreclr version

```c#
public class Program
{
    public class Test3 {
        public string Title {get;set;}
    }

    public void Main(string[] args)
    {
        string xml = "<?xml version=\"1.0\" encoding=\"utf-8\"?><Test1 xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\" xmlns:xsd=\"http://www.w3.org/2001/XMLSchema\"><Title>Test 12345</Title></Test3>";
        XmlSerializer serializer = new XmlSerializer(typeof(Test3));
        StringReader reader = new StringReader(xml);
        var value = (Test3)serializer.Deserialize(reader);
        var title = value.Title;

    }
 }

```

the stacktrace:
```
System.InvalidOperationException: There is an error in XML document (0, 0). ---> System.NotSupportedException: Specified method is not supported.
   at System.Xml.Serialization.XmlSerializerImplementation.get_ReadMethods()
   at System.Xml.Serialization.TempAssembly.InvokeReader(XmlMapping mapping, XmlReader xmlReader, Object events, String encodingStyle)
   at System.Xml.Serialization.XmlSerializer.Deserialize(XmlReader xmlReader, String encodingStyle, Object events)
   --- End of inner exception stack trace ---
   at System.Xml.Serialization.XmlSerializer.Deserialize(XmlReader xmlReader, String encodingStyle, Object events)
   at System.Xml.Serialization.XmlSerializer.Deserialize(TextReader textReader)
   at SampleApp.Program.Main(String[] args)
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at Microsoft.Framework.Runtime.Common.EntryPointExecutor.Execute(Assembly assembly, String[] args, IServiceProvider serviceProvider)
   at Microsoft.Framework.ApplicationHost.Program.ExecuteMain(DefaultHost host, String applicationName, String[] args)
   at Microsoft.Framework.ApplicationHost.Program.Main(String[] args)
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at Microsoft.Framework.Runtime.Common.EntryPointExecutor.Execute(Assembly assembly, String[] args, IServiceProvider serviceProvider)
   at dnx.host.Bootstrapper.RunAsync(List`1 args, IRuntimeEnvironment env)
   at dnx.host.RuntimeBootstrapper.ExecuteAsync(String[] args)
   at dnx.host.RuntimeBootstrapper.Execute(String[] args)
System.NotSupportedException: Specified method is not supported.
   at System.Xml.Serialization.XmlSerializerImplementation.get_ReadMethods()
   at System.Xml.Serialization.TempAssembly.InvokeReader(XmlMapping mapping, XmlReader xmlReader, Object events, String encodingStyle)
   at System.Xml.Serialization.XmlSerializer.Deserialize(XmlReader xmlReader, String encodingStyle, Object events)
```

**Comment by jasonwilliams200OK** [comment]
I found these two related PRs: https://github.com/dotnet/corefx/pull/1322 & https://github.com/dotnet/corefx/pull/1917. Apparently CI build is passing all tests on Linux with current master: http://dotnet-ci.cloudapp.net/job/dotnet_corefx_linux_release/686/console. Are you testing with latest changes?

**Comment by stephentoub** [comment]
I just tried this with the current src in corefx.  Your example fails, on both Windows and Linux, because it actually is invalid XML: the opening tag is for "Test1" while the closing tag is for "Test3".  When I fix both to be "Test3", it successfully deserializes on both Windows and Linux.

**Comment by stephentoub** [comment]
@IRooc, I'm going to close this out, but please do reopen if you're able to repro with the current code.

**Comment by IRooc** [comment]
@stephentoub  Sorry I was AFK for the weekend. There was a small Typo in the inline xml string it should read: Test3 instead of Test1 in the example above:
```
    public class Program
	{
		public class Test3
		{
			public string Title { get; set; }
		}
		public void Main(string[] args)
        {
			string xml = "<?xml version=\"1.0\" encoding=\"utf-8\"?><Test3 xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\" xmlns:xsd=\"http://www.w3.org/2001/XMLSchema\"><Title>Test 12345</Title></Test3>";
			XmlSerializer serializer = new XmlSerializer(typeof(Test3));
			StringReader reader = new StringReader(xml);
			var value = (Test3)serializer.Deserialize(reader);
			var title = value.Title;
		}
    }
```

THis works fine on windows but not on linux ubuntu with  1.0.0-beta6-12120 -r coreclr -arch x64

**Comment by stephentoub** [comment]
@khdang, can you take a look?

**Comment by khdang** [comment]
@IRooc With the latest code, the issue works fine on both Windows and Linux for me. All tests on CI are passing as well. Are you still having this issue?

**Comment by IRooc** [comment]
fixed in ```1.0.0-beta8-15654```
